### ISSUE TITLE
Weird result

### ISSUE THREAD

**Issue by HungNgoCT** [graphdeco-inria/hierarchical-3d-gaussians#61]
Hi.,

Thank you for method and your open source.

I tried to repeat your steps on my data to understand as followings:
- Used my phone to capture an elephant model with 3 rounds.
- Extracted frames with 1 frame per second with the name 0001.jpg, 0002.jpg, 0003.jpg... as in the first 2 images
- Then, created 2 folder cam1 include images 0001.jpg, 0003.jpg, 0005.jpg... and cam2: 0002.jpg. 0004.jpg, 0006.jpj...
- All the steps were fine without error. But I got weird results as in the video
Note that I did experiment on sample data, results are fine. Also, I used 3D GS, result was also fine for my data.

Can anyone help me to understand this? I also share the link of my data here.
https://drive.google.com/file/d/14UgqhZp6FQFLKmOZVv-DV98Q79CoeQNv/view?usp=sharing

Thank you

![0002](https://github.com/user-attachments/assets/e9447c08-d01b-4534-b35c-377fe19f7acf)
![0006](https://github.com/user-attachments/assets/3bad9c30-5df8-4189-bc1d-7736db784e4d)

https://github.com/user-attachments/assets/c5ebf1ca-463e-4fdb-994f-903450c4fb1e

**Comment by ameuleman** [comment]
Hi,
Did you use the same calibration to optimize 3DGS and H3DGS? Could you please visualize COLMAP results?
Can you snap to the closest camera in the viewer?
Also, if images are ordered and captured from the same camera, they should be in the same subfolder

**Comment by HungNgoCT** [comment]
Hi @ameuleman ,
Thank for reply. I checked, the colmap results seem correct (I change the code to use exhaustive matching and mapper). Colmap failed when use hierarchical_mapper
![image](https://github.com/user-attachments/assets/2dfa07b0-7ed3-4eb5-b967-4f2dfbf6fbd6)

**Comment by HungNgoCT** [comment]
Thank you @ameuleman ,

Problem because number of my images is less than 100 while default setting for min no.  of chunk images is 100. It cannot create chunks.

But anyway, result is not nice as conventional GS. Do you have any advise?

https://github.com/user-attachments/assets/3183fc5e-8fcb-4505-9df3-df642e719bf4

**Comment by ameuleman** [comment]
Our method is tuned (learning rates and parameters) for larger scenes with more supervising images, we could expect some degradation on small scenes 3DGS is tuned for.
Running the single-chunk optimization only should work better (`images/` and `depths/` paths should be set appropriately):
```
python train_single.py -s ${CHUNK_DIR} --model_path ${OUTPUT_DIR} -d depths --skip_scale_big_gauss
submodules/gaussianhierarchy/build/GaussianHierarchyCreator ${OUTPUT_DIR}/point_cloud/iteration_30000/point_cloud.ply ${CHUNK_DIR}  ${OUTPUT_DIR} 
python train_post.py -s ${CHUNK_DIR} --model_path ${OUTPUT_DIR} --hierarchy ${OUTPUT_DIR}/hierarchy.hier --iterations 15000 --feature_lr 0.0005 --opacity_lr 0.01 --scaling_lr 0.001
```

**Comment by HungNgoCT** [comment]
Hi @ameuleman ,

Thank you for your further guidance.

I tried to follow your guidance on training for single trunk. However, it usually cause "long Gaussians" as shown in the image below, while "long Gaussians" do not appear in 3D GS. Could you please predict the reasons? Or, what parameters I should adjust to reduce this? Thanks

![image](https://github.com/user-attachments/assets/b640b02f-d28a-42f0-b2b9-513ada4f2f55)

**Comment by Linkersem** [comment]
hi, i also have meet same problem,The quality of H3DGS is relatively poor compared to 3DGS, and there are more burrs, as you mentioned “long Gaussians”.

**Comment by ameuleman** [comment]
Hi,
Are they run on the same calibration and use the same SfM points? 
Would you mind sharing the calibrated data for this scene?

**Comment by HungNgoCT** [comment]
> Hi, Are they run on the same calibration and use the same SfM points? Would you mind sharing the calibrated data for this scene?

Hi @ameuleman ,

I used the same steps as guidance, just revised source  code in order to use exhaustive matching, and min number of chunk images is 50 instead of 100:
1) Run generate_colmap
2) Run generate_chunks
3) Run generate_depth
4) Run full_train

All steps completed without any errors

Data, camera_calibration, output are here.
https://drive.google.com/file/d/1oUrUfDkRKBm67ty1HMUj066yP_2EfA1U/view?usp=sharing

**Comment by ameuleman** [comment]
Hi, Could you please share the output for chunk 1_0 (where I assume the capture above is from) or the merged hierarchy?

**Comment by ameuleman** [comment]
When rendering from chunk 0_0's hierarchy.hier_opt, I do not observe strong artefacts. Are you using a compatible renderer when visualizing the scene? Visualizing a ply file with a viewer that does not support correct antialiasing can lead to similar artefacts (see https://github.com/graphdeco-inria/hierarchical-3d-gaussians/issues/47).

**Comment by HungNgoCT** [comment]
> When rendering from chunk 0_0's hierarchy.hier_opt, I do not observe strong artefacts. Are you using a compatible renderer when visualizing the scene? Visualizing a ply file trained with correct antialiasing with a viewer that does not support it leads to similar artefacts (see #47).

Hi @ameuleman ,

The link I share includes all the data. The output of the chunks are in output\trained_chunks\XX.

I used SIBR_gaussianViewer_app.exe from 3DGS to render for each chunk. Is it correct to use this? The render as image below
![image](https://github.com/user-attachments/assets/4cf5e723-9a66-41c2-b9be-4f0941b07ba8)

**Comment by Linkersem** [comment]
hi, did you use the new SIBR_viewer for the output of h3dgs ?

**Comment by HungNgoCT** [comment]
> hi, did you use the new SIBR_viewer for the output of h3dgs ?

I used new SIBR_viewer to see final result. For each chunk, I use viewer from 3DGS

**Comment by Linkersem** [comment]
Emm,sry .What I mean is that the final model file with the suffix name .hier is the model file finally output by h3dgs. The new viewer cannot view a single ply file, although the new SIBR_viewer project file contains the code for rendering the ply file ( This code is used to render the original 3dgs ply file), but it is not compiled into the final binary, you can try to do this if you want.

**Comment by HungNgoCT** [comment]
> Emm,sry .What I mean is that the final model file with the suffix name .hier is the model file finally output by h3dgs. The new viewer cannot view a single ply file, although the new SIBR_viewer project file contains the code for rendering the ply file ( This code is used to render the original 3dgs ply file), but it is not compiled into the final binary, you can try to do this if you want.

Thank you @Linkersem . I will give a try with this.

**Comment by Linkersem** [comment]
@ameuleman Hello, no, I used the new SIBR_viewer to render h3dgs training results. This is what I trained.This is a screenshot of part of my training. You can see that there are a lot of burrs and it is more obvious in the font .The equipment I used to collect data cannot do the same as the one in the paper, so the density I collected is relatively dense.
<img width="254" alt="0ed22040e193f73fcfaf96962975279" src="https://github.com/user-attachments/assets/cdb9587e-e670-4a56-b18e-7e6195d522ad">
![image](https://github.com/user-attachments/assets/ce624f45-4230-4819-b52e-e8cc357cb3be)

**Comment by ameuleman** [comment]
Could poses be off? Do you get much better results with 3DGS trained on a single chunk?

**Comment by HungNgoCT** [comment]
> @ameuleman Hello, no, I used the new SIBR_viewer to render h3dgs training results. This is what I trained.This is a screenshot of part of my training. You can see that there are a lot of burrs and it is more obvious in the font .The equipment I used to collect data cannot do the same as the one in the paper, so the density I collected is relatively dense. <img alt="0ed22040e193f73fcfaf96962975279" width="254" src="https://private-user-images.githubusercontent.com/49711774/368867880-cdb9587e-e670-4a56-b18e-7e6195d522ad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3Mjg1MDEsIm5iZiI6MTcyNjcyODIwMSwicGF0aCI6Ii80OTcxMTc3NC8zNjg4Njc4ODAtY2RiOTU4N2UtZTY3MC00YTU2LWIxOGUtN2U2MTk1ZDUyMmFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE5VDA2NDMyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4MTM5NGNjZGE0YjU0ZjMwMjBkODI4ZWE3ZWRiZDIxZmFhMmQ1OTJmYjk1MTU2OWEyZjY4ZGMxZjM4MWEzNGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XXpKS1vxJRLm8fg_zncRUTz4FHejs7QBGUcMcRJ2YJc"> ![image](https://private-user-images.githubusercontent.com/49711774/368868506-ce624f45-4230-4819-b52e-e8cc357cb3be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3Mjg1MDEsIm5iZiI6MTcyNjcyODIwMSwicGF0aCI6Ii80OTcxMTc3NC8zNjg4Njg1MDYtY2U2MjRmNDUtNDIzMC00ODE5LWI1MmUtZThjYzM1N2NiM2JlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE5VDA2NDMyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiOTczNWI0ZmYwMzhjMDcwZjM3YjhiZDQ2MzJlNTEyZjAwMDdjMGRkMGE1ZmIwMmUwZGE3ZjY1YmQxY2ViYWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HdbPamKoOWK6Xisyo2f_R_vHs4Cgv4d9zb6Y_V-B6PI)

Hi @ameuleman .,

I have the same situation with this. There are "burrs", specially near edges. This phenomenon  does not have when using 3DGS.

**Comment by ameuleman** [comment]
Training H3DGS and 3DGS on the same chunk and COLMAP will help us isolate the issue. If both methods show similar artefacts, we should look into the preprocessing and chunk generation scripts, if not, we should look into the single-chunk optimization

**Comment by Linkersem** [comment]
Okay, I will use h3dgs to train this data as a chunk and give feedback. Once the training is completed, I will post my training results.I would like to ask whether this phenomenon may be caused by the change of densify strategy? Or can I optimize the effect by adjusting parameters like learning rate?

**Comment by ameuleman** [comment]
Thank you
If it is related to the optimization, I could see  two main differences that could, in some cases, harm results: learning rates tuned for larger and sparser scenes or the depth supervision giving wrong signals (the monocular depth estimator might struggle in some scenes or the depth scale might not be correctly estimated)

**Comment by Linkersem** [comment]
Hi, @ameuleman ,I greatly modified chunk_size to control that only one chunk is generated. Unfortunately, I got similar results as before, as shown in the figure below.
![image](https://github.com/user-attachments/assets/4291dfb8-ca4f-449a-88e4-e667f7c0a103)
![image](https://github.com/user-attachments/assets/470542fe-6790-431b-92ff-8f98e3ae3050)

 In addition, for the same scene, I have another set of sparse data, but the fonts of this set of data are much clearer. There are a lot of holes in the whole scene and the scene is worse. I wonder if it is because a lot of points were deleted by triangulation and ba when making make_chunk.
![image](https://github.com/user-attachments/assets/d389546c-6680-49c2-bea4-acb86b0154a2)
![image](https://github.com/user-attachments/assets/3239e50d-31e7-4408-910a-8dfa7b146139)

**Comment by ameuleman** [comment]
Hi, 
Did you run h3dgs single chunk optimisation (train_single.py) and 3dgs on the same chunk?
There should be no need to regenerate chunk to test this

**Comment by Linkersem** [comment]
ohh,my bad, I understood it wrong. I will train this block separately and give a screenshot comparison of the results of 3dgs and h3dgs.

**Comment by Linkersem** [comment]
Hi, @ameuleman I have trained my single chunk sparse data through h3dgs and 3dgs respectively, both results are poor, and the results of 3dgs seem to be worse.
This is the result of H3DGS training
![h3dgs_A5chun10new](https://github.com/user-attachments/assets/fa77c560-1464-4c40-b1a1-df4de28a0982)
this is the result of 3dgs
![3dgs_A5chun10new](https://github.com/user-attachments/assets/04a52db4-20fc-46fb-b033-e3d743c1030b)
Thank you for your patience all the time.

**Comment by ameuleman** [comment]
Thank you. This means that the issue comes from preprocessing. 
* COLMAP might struggle to estimate poses (when making the global calibration or during per-chunk bundle adjustment)
* The chunk generation might have issues (too few images per chunk, too many images dropped or the chunk might be too large)

**Comment by ameuleman** [comment]
The COLMAP screenshot you shared is not aligned. Is it the one used to generate chunks?

**Comment by Linkersem** [comment]
Hi,the colmap screenshot I provided earlier is not this chunk's, the colmap screenshot of this chunk is as follows:
![image](https://github.com/user-attachments/assets/f35609a1-6090-4ce0-9f57-9a8200b4e1a6)

And its z-axis doesn't seem to be facing upwards either,I looked at the sparse file in the align folder and its coordinate system is the same as this image.Is there a problem with any of the steps?

**Comment by ameuleman** [comment]
What does the full COLMAP look like? There might be outlier poses that break the auto aligner

**Comment by Linkersem** [comment]
The full is this ,and for clarity, I enlarged it and took two screenshots. 
![image](https://github.com/user-attachments/assets/19c64eb9-c766-4a70-955f-7a1f0bc0151b)

![image](https://github.com/user-attachments/assets/1e530091-4fea-401a-93ae-6aba7d4998e2)
![image](https://github.com/user-attachments/assets/e717af30-99d6-4b0c-8320-17d5999dab98)

**Comment by Linkersem** [comment]
Do you need the colmap file in the rectified folder? I can send it to your email hhhh.

**Comment by ZERO-black** [comment]
Hi, I also encountered the “long Gaussians” issue in H3DGS, which doesn’t appear in 3DGS when using the same data and calibration.

Since the discussion has been quiet for a while, I was wondering if there have been any new insights or if this remains a known issue with H3DGS?
